src/Tes.Runner/Docker/DockerExecutor.cs

```diff
@@ -186,35 +187,16 @@ private async Task PullImageWithRetriesAsync(string imageName, string? tag, Auth
                     new Progress<JSONMessage>(message => logger.LogDebug(message.Status))));
         }
 
-        private async Task DeleteAllImagesAsync()
+        private async Task DeleteImageAsync(string imageName)
         {
             try
             {
-                var images = await dockerClient.Images.ListImagesAsync(new ImagesListParameters { All = true });
-
-                if (images is null)
-                {
-                    return;
-                }
-
-                foreach (var image in images)
-                {
-                    try
-                    {
-                        await dockerClient.Images.DeleteImageAsync(image.ID, new ImageDeleteParameters { Force = true });
-                        logger.LogInformation(@"Deleted Docker image with ID: {ImageID}", image.ID);
-                    }
-                    catch (Exception e)
-                    {
-                        logger.LogInformation(@"Failed to delete image with ID: {ImageID}. Error: {ErrorMessage}", image.ID, e.Message);
-                        throw;
-                    }
-                }
+                await dockerClient.Images.DeleteImageAsync(imageName, new ImageDeleteParameters { Force = true });
```

Comment:
does this work as expected?  I thought it needed the image.ID, which I didn't think was the same as `imageName`

Reply:
Yeah it works, the trick was to get the container object and then use that to get the image name. The delete api will accept either the image sha or an image name with tag, however if you try executionOptions.ImageName it won't work sometimes for images with a "latest" tag. When you pull it dcoker does the substitution from ubuntu:latest to ubuntu:22.04 so you can't delete by calling delete on ubuntu:latest. (I would think it should keep both tags, but it doesn't seem to via the api at least)